lib/oppen/mixins.rb

```diff
@@ -21,30 +21,50 @@ def upsize_circular_array(arr, offset)
     #
     # @param tokens [Array[Token]]
     # @param base_indent [Integer]
+    # @param printer_name [String]
     #
     # @return [String]
-    def tokens_to_wadler(tokens, base_indent = 4)
+    def tokens_to_wadler(tokens, base_indent: 0, printer_name: 'out')
+      nb_spaces = base_indent
       out = StringIO.new
-      write = ->(txt, nb_spaces) {
-        out.write("#{' ' * nb_spaces}#{txt}\n")
+
+      write = ->(txt, indentation) {
+        out.write("#{' ' * indentation}#{txt}\n")
```

Comment:
I would simply 

```
out << ' ' * indentation << txt << "n"
```

which is more readable, and certainly faster.

Also, I would simply capture `indentation`/`nb_spaces` instead of having to pass it as an argument, since you are always passing exactly `nb_spaces`.